components/providers/dashboard-event-edition-context-provider.tsx

```diff
@@ -0,0 +1,185 @@
+"use client";
+
+import { useAuth } from "@clerk/nextjs";
+import {
+  useSuspenseInfiniteQuery,
+  useSuspenseQuery,
+} from "@tanstack/react-query";
+import {
+  createContext,
+  useContext,
+  useMemo,
+  useRef,
+  useTransition,
+} from "react";
+import { useForm } from "react-hook-form";
+import { useTranslations } from "next-intl";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Form } from "../shadcn/form";
+import { EventInfo } from "@/app/gen/zenao/v1/zenao_pb";
+import { makeLocationFromEvent } from "@/lib/location";
+import { useEditEvent } from "@/lib/mutations/event-management";
+import {
+  communitiesListByEvent,
+  communityIdFromPkgPath,
+  DEFAULT_COMMUNITIES_LIMIT,
+} from "@/lib/queries/community";
+import { eventGatekeepersEmails, eventOptions } from "@/lib/queries/event";
+import { EventUserRole, eventUserRoles } from "@/lib/queries/event-users";
+import { userInfoOptions } from "@/lib/queries/user";
+import { eventFormSchema, EventFormSchemaType } from "@/types/schemas";
+import { useToast } from "@/hooks/use-toast";
+import { captureException } from "@/lib/report";
+import { useAnalyticsEvents } from "@/hooks/use-analytics-events";
+
+interface DashboardEventEditionContextProps {
+  roles: EventUserRole[];
+  eventId: string;
+  eventInfo: EventInfo;
+  isUpdating: boolean;
+  isSubmittable?: boolean;
+  formRef?: React.RefObject<HTMLFormElement | null>;
+  save: (values: EventFormSchemaType) => void;
+}
+
+const DashboardEventEditionContext =
+  createContext<DashboardEventEditionContextProps>({} as never);
+
+interface DashboardEventEditionContextProviderProps {
+  eventId: string;
+  children: React.ReactNode;
+}
+
+export function useDashboardEventEditionContext() {
+  const context = useContext(DashboardEventEditionContext);
+
+  if (!context) {
+    throw new Error(
+      "useDashboardEventEditionContext must be used within a DashboardEventEditionContextProvider",
+    );
+  }
+  return context;
+}
+
+export default function DashboardEventEditionContextProvider({
+  eventId,
+  children,
+}: DashboardEventEditionContextProviderProps) {
+  const t = useTranslations("eventForm");
+  const { toast } = useToast();
+  const { trackEvent } = useAnalyticsEvents();
+  const { getToken, userId } = useAuth();
+  const { editEvent } = useEditEvent(getToken);
+  const { data: eventInfo } = useSuspenseQuery(eventOptions(eventId));
```

Comment:
what about using useDashboardEventContext for some data here ? in the layout.tsx it's the parent and it fetches similar data (roles, eventInfo) isn't ?